Projects/Features/SearchFeature/Sources/ViewControllers/BeforeSearchContentViewController.swift

```diff
@@ -11,32 +12,52 @@ import RxSwift
 import UIKit
 import Utility
 
-protocol BeforeSearchContentViewDelegate: AnyObject {
-    func itemSelected(_ keyword: String)
-}
+// DatSource model 실제 Entity로 교체
 
-public final class BeforeSearchContentViewController: BaseStoryboardReactorViewController<BeforeSearchReactor> {
-    @IBOutlet weak var tableView: UITableView!
-    @IBOutlet weak var indicator: NVActivityIndicatorView!
+public final class BeforeSearchContentViewController: BaseReactorViewController<BeforeSearchReactor> {
+    fileprivate enum Section: Int {
+        case top
+        case mid
+        case bottom
+    }
 
-    weak var delegate: BeforeSearchContentViewDelegate?
+    enum DataSource: Hashable {
+        case youtube(model: Model)
+        case recommand(model2: Model)
+        case popularList(model: Model)
+
+        var title: String {
+            switch self {
+            case let .youtube(model):
+                return model.title
+            case let .recommand(model2):
+                return model2.title
+            case let .popularList(model):
+                return model.title
+            }
+        }
+    }
+
+    fileprivate var playlistDetailFactory: PlaylistDetailFactory!
+    fileprivate var textPopUpFactory: TextPopUpFactory!
```

Comment:
```suggestion
    fileprivate let playlistDetailFactory: PlaylistDetailFactory
    fileprivate let textPopUpFactory: TextPopUpFactory
```
init에서 완전 초기화 된다는게 확정되니 let이랑 강제 언래핑도 빼도 좋겠네요

Reply:
반영했습니다.